vowpalwabbit/OjaNewton.cc

```diff
@@ -52,51 +52,53 @@ struct OjaNewton {
     bool normalize;
     bool random_init;
 
-    void initialize_Z()
-    {
-        size_t stride_shift = all->reg.stride_shift;
-        weight* weights = all->reg.weight_vector;
-
-        uint32_t length = 1 << all->num_bits;
-        if(normalize) { // initialize normalization part
-	    for (uint32_t i = 0; i < length; i++)
-	        weights[(i << stride_shift) + NORM2] = 0.1;
-        }
-
+	void initialize_Z()
+	{  weight_vector& weights = all->wv;
+       uint32_t length = 1 << all->num_bits;
+	   if (normalize) { // initialize normalization part
+		 for (weight_vector::iterator i = weights.begin(NORM2); i != weights.end(NORM2); ++i)
+		   *i = 0.1;
+	    }
         if(!random_init) {
             // simple initialization
-            for (int i = 1; i <= m; i++)
-                weights[(i << stride_shift) + i] = 1;
+		  weight_vector::iterator iter = weights.begin(0) + 1;
+		  for (int i = 1; i <= m; ++i, ++iter)
+		  { weight_vector::iterator::w_iter j = iter.begin() + i;
+			*j = 1;
+		  }
         }
```

Comment:
If you rewrote the code for your iterator as I suggested, again, you wouldn't need the loop with the (single letter name variable! :D) i.  This would simply end up being a std::fill or `boost::fill(*iter, 1)`.